src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDashboard/requestLog.tsx

```diff
@@ -0,0 +1,331 @@
+import React from 'react';
+import styled from 'react-emotion';
+import moment from 'moment-timezone';
+
+import AsyncComponent from 'app/components/asyncComponent';
+
+import {Panel, PanelBody, PanelHeader, PanelItem} from 'app/components/panels';
+import EmptyMessage from 'app/views/settings/components/emptyMessage';
+import DateTime from 'app/components/dateTime';
+import DropdownControl, {DropdownItem} from 'app/components/dropdownControl';
+import Tag from 'app/views/settings/components/tag';
+import Link from 'app/components/links/link';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import Checkbox from 'app/components/checkbox';
+import Button from 'app/components/button';
+
+import theme from 'app/utils/theme';
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import {SentryApp, SentryAppWebhookRequest} from 'app/types';
+
+const ALL_EVENTS = t('All Events');
+const MAX_PER_PAGE = 10;
+
+const ResponseCode = ({code}: {code: number}) => {
+  let priority = 'error';
+  if (code <= 399 && code >= 300) {
+    priority = 'warning';
+  } else if (code <= 299 && code >= 100) {
+    priority = 'success';
+  }
+
+  return (
+    <div>
+      <Tag priority={priority}>{code === 0 ? 'timeout' : code}</Tag>
+    </div>
+  );
+};
+
+const TimestampLink = ({date, link}: {date: moment.MomentInput; link?: string}) => {
+  return link ? (
+    <Link to={link}>
+      <DateTime date={date} />
+    </Link>
+  ) : (
+    <DateTime date={date} />
+  );
+};
+
+type Props = AsyncComponent['props'] & {
+  app: SentryApp;
+};
+
+type State = AsyncComponent['state'] & {
+  eventType: string;
+  errorsOnly: boolean;
+  currentPage: number;
+  requests: SentryAppWebhookRequest[];
+};
+
+export default class RequestLog extends AsyncComponent<Props, State> {
+  shouldReload = true;
+
+  get eventTypes() {
+    // TODO(nola): ideally this would be kept in sync with EXTENDED_VALID_EVENTS on the backend
+
+    const {app} = this.props;
+    let events = [ALL_EVENTS, 'installation.created', 'installation.deleted'];
+
+    if (app.events.includes('error')) {
+      events = [...events, 'error.created'];
+    }
+    if (app.events.includes('issue')) {
+      events = [
+        ...events,
+        'issue.created',
+        'issue.resolved',
+        'issue.ignored',
+        'issue.assigned',
+      ];
+    }
+    if (app.isAlertable) {
+      events = [...events, 'event_alert.triggered'];
+    }
+
+    const issueLinkComponent: any = (app.schema.elements || []).find(
+      (element: any) => element.type === 'issue-link'
```

Comment:
@nolachen I would just change the type of `SentryApp.schema.elements` to any[] so you can avoid typecasting. Or if you really want to go the extra mile, you could define the schema of the elements based on how you are using them (it's OK if the type is incomplete).